selfservice/flow/registration/error.go

```diff
@@ -115,7 +117,38 @@ func (s *ErrorHandler) WriteFlowError(
 	}
 
 	if f.Type == flow.TypeBrowser && !x.IsJSONRequest(r) {
-		http.Redirect(w, r, f.AppendTo(s.d.Config().SelfServiceFlowRegistrationUI(r.Context())).String(), http.StatusFound)
+		isWebView, innerErr := flow.IsWebViewFlow(r.Context(), s.d.Config(), f)
+		if innerErr != nil {
+			s.forward(w, r, f, innerErr)
+			return
+		}
+
+		var redirectLocation = ""
+		if isWebView {
+			c := s.d.Config()
+			returnTo, innerErr := x.SecureRedirectTo(r, c.SelfServiceBrowserDefaultReturnTo(r.Context()),
+				x.SecureRedirectUseSourceURL(f.RequestURL),
+				x.SecureRedirectAllowURLs(c.SelfServiceBrowserAllowedReturnToDomains(r.Context())),
+				x.SecureRedirectAllowSelfServiceURLs(c.SelfPublicURL(r.Context())),
+				x.SecureRedirectOverrideDefaultReturnTo(s.d.Config().SelfServiceFlowLoginReturnTo(r.Context(), f.Active.String())),
+			)
+			if innerErr != nil {
+				s.forward(w, r, f, innerErr)
+				return
+			}
+
+			query := returnTo.Query()
+
+			query.Set("code", strconv.Itoa(int(f.UI.Messages[0].ID)))
```

Comment:
Are we sure `f.UI.Messages[0]` exists ?

Reply:
Have added check for messages array length.